### PR TITLE
XRENDERING-672: Provide a way to list the categories of a macro

### PR DESCRIPTION
**Pull request by manuelleduc**
Introduce `MacroCategoryManager#getMacroCategories(org.xwiki.rendering.macro.MacroId)` and its implementation `DefaultMacroCategoryManager#getMacroCategories(org.xwiki.rendering.macro.MacroId)`

Jira: https://jira.xwiki.org/browse/XRENDERING-672

### CODE REVIEW

File: xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManager.java [hunk 1]
```diff
@@ -161,4 +186,9 @@ private void addToCategory(Map<String, Set<MacroId>> result, MacroId macroId, St
         ids.add(macroId);
         result.put(macroCategory, ids);
     }
+
+    private Set<String> splitCategories(String s)
+    {
+        return Arrays.stream(s.split("\\s*,\\s*")).collect(Collectors.toSet());
```
Comment (michitux) [inline xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManager.java:192]: If you really want/need to use a regular expression here, this expression should be pre-compiled as a static variable (this should also produce a warning in SonarLint). Further, why aren't you using our standard list parsing implementation that we normally use for configuration variables?

Also, I think this discards the order of the categories (or at least doesn't guarantee it), is this intentional?

File: xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/test/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManagerTest.java [hunk 2]
```diff
@@ -19,122 +19,181 @@
...
-import org.xwiki.rendering.syntax.Syntax;
 import org.xwiki.rendering.transformation.macro.MacroTransformationConfiguration;
-import org.xwiki.test.jmock.AbstractComponentTestCase;
+import org.xwiki.test.LogLevel;
+import org.xwiki.test.junit5.LogCaptureExtension;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectMockComponents;
+import org.xwiki.test.junit5.mockito.MockComponent;
+
+import ch.qos.logback.classic.Level;
```
Comment (michitux) [inline xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/test/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManagerTest.java:41]: Is this really the correct import?

File: xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/test/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManagerTest.java [hunk 3]
```diff
@@ -19,122 +19,181 @@
...
+        when(this.macroManager.getMacroIds()).thenReturn(Set.of(macroAId));
+        when(this.macroManager.getMacro(macroAId)).thenReturn(macroA);
+        when(macroA.getDescriptor()).thenReturn(macroADescriptor);
 
-        // Check for overwritten categories.
-        Assert.assertTrue(macroCategories.contains("Content"));
-        Assert.assertTrue(macroCategories.contains("Simple"));
+        // Check whether our macros are registered under correct categories.
+        Set<Object> expected = new HashSet<>();
+        expected.add(null);
```
Comment (michitux) [inline xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/test/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManagerTest.java:100]: Why do we expect a set with `null`? Wouldn't an empty set be better?

**Review by michitux — CHANGES_REQUESTED** [review]

File: xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/test/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManagerTest.java (hunk 2 above)
Comment (manuelleduc) [reply xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/test/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManagerTest.java:41]: Yes, that's the type returned by `ch.qos.logback.classic.spi.ILoggingEvent#getLevel` (`ch.qos.logback.classic.Level`)

File: xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/test/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManagerTest.java (hunk 3 above)
Comment (manuelleduc) [reply xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/test/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManagerTest.java:100]: I've kept the logic where a macro without define category was affected to the "null" category. Without this special category, macros without category wouldn't be returned by some higher level operations.

File: xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManager.java (hunk 1 above)
Comment (manuelleduc) [reply xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManager.java:192]: Sonar is not raising an error (at least not in my ide), but indeed per-compilling the regex is a good idea.
`MacroTransformationConfiguration#getCategories` returns a `java.util.Properties` and not a `commons-configuration` Properties object. And it does not have the notion of list.

**Pull request merged**